.github/workflows/update-leafygreen-packages.yaml

```diff
@@ -0,0 +1,54 @@
+name: Update leafygreen packages
+
+# Runs nightly and manually
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  update_generated_files:
+    name: Update Leafygreen
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # don't checkout a detatched HEAD
+          ref: ${{ github.head_ref }}
+
+          # this is important so git log can pick up on
+          # the whole history to generate the list of AUTHORS
+          fetch-depth: '0'
+
+      - name: Setup git
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20.16.0
+          cache: 'npm'
+
+      - name: Install npm@10.2.4
+        run: |
+          npm install -g npm@10.2.4
+      - name: Install Dependencies
+        run: |
+          npm -v
+          npm ci
+      - name: Bump packages
+        run: |
+          npm add $(cat packages/compass-components/package.json | jq -r '.dependencies | keys[]' | grep "@leafygreen-ui" | sed s/$/@latest/ | xargs)  --workspace=@mongodb-js/compass-components
```

Comment:
Copied from https://github.com/10gen/compass-data-explorer/#publishing-packages

Reply:
now i am thinking about `jq` - would it be available on CI 🤔 ?

Reply:
and its available there: https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#tools